File: DataStructure_HandsOn/Java/Program15.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program15:
//		A program which displays Largest digits of all elements from singly linear linked list.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class SinglyLL
{
	static void DisplayLargest(LinkedList obj)
	{
		
		System.out.print("Largest Digits: ");
		Iterator <Integer>iObj = obj.iterator();
		while(iObj.hasNext())
		{
			int iDigit = iObj.next();
			int iRem = 0,iRev=0,iMax=0;
			
			while(iDigit!=0)
			{
				iRem = iDigit%10;
				if(iMax<iRem)
				{
					iMax = iRem;
				}
				iDigit = iDigit/10;
			}
			
			System.out.print(iMax+" ");
		}

		System.out.println();
	}

}

class Program15
{
	public static void main(String[] args)
	{
		LinkedList <Integer>lObj = new LinkedList<Integer>();
		int arr[] = new int[]{11,20,32,41};

		for(int i=0;i<4;i++)
		{
			lObj.add(arr[i]);
		}

		Iterator <Integer>iObj = lObj.iterator();
		System.out.print("Input Linked List: [");
		while(iObj.hasNext())
		{
			System.out.print(iObj.next()+", ");
		}

		System.out.println("]");
		
		SinglyLL.DisplayLargest(lObj);		
		
	}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//		Output:
//			Input Linked List: [11, 20, 32, 41, ]
//			Largest Digits: 1 2 3 4 
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

File: DataStructure_HandsOn/C++/program15.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program15:
//		A program which displays Largest digits of all elements from singly linear linked list.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef struct node
{
	int data;
	struct node* next;

}NODE,*PNODE;

class SinglyLL
{
	private:
		PNODE first;
		int no;
	
	public:
		SinglyLL(PNODE first=NULL,int no=0)
		{
		
		}

		void InsertFirst(int no)
		{
			PNODE newn = new NODE;

			newn->data = no;
			newn->next = NULL;

			if(first == NULL)
			{
				first = newn;
			}
			else
			{
				newn->next = first;
				first = newn;
			}
		}	

		void DisplayLargest()
		{
			PNODE Head = first;
			while(Head != NULL)
			{
				int iDigit = Head->data;
				int iRem = 0,iRev = 0,iMax=0;
				while(iDigit!=0)
				{
					iRem = iDigit%10;
					if(iMax < iRem)
					{
						iMax = iRem;
					}	
					iDigit = iDigit/10;

				}
		
				cout<<iMax<<" ";
				Head = Head->next;
			}
			cout<<"\n";

		}

		void Display()
		{
			PNODE temp = first;

			cout<<"Input Linked List: ";
			while(temp!=NULL)
			{
				cout<<"|"<<temp->data<<"|->";
				temp=temp->next;
			}
			cout<<"NULL\n";
		}
};


int main()
{
	int i = 0,iRet = 0;
	int arr[4] = {419,532,250,11};

	SinglyLL *sObj = new SinglyLL();
	

	for(i=0;i<4;i++)
	{
		sObj->InsertFirst(arr[i]);
	}	
	sObj->Display();
	
	sObj->DisplayLargest();

	return 0;
}
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Output:
//		Input Linked List: |11|->|250|->|532|->|419|->NULL
//		1 5 5 9 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
